src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,841 @@ static DamageTable D_80AD2550 = {
+        return;
+    }
+    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        ActorCutscene_Stop(0x7C);
+    }
+    ActorCutscene_SetIntentToPlay(this->cutscene);
+}
+
+void func_80AD16A8(EnOsn* this, GlobalContext* globalCtx) {
+    u8 pad;
+    u32 temp_v0;
+
+    if (func_800EE29C(globalCtx, 0x82)) {
+        temp_v0 = func_800EE200(globalCtx, 0x82);
+        this->unk_1F0 = 0;
+        if (this->unk_1ED != globalCtx->csCtx.npcActions[temp_v0]->unk0) {
+            this->unk_1ED = globalCtx->csCtx.npcActions[temp_v0]->unk0;
+            switch (globalCtx->csCtx.npcActions[temp_v0]->unk0) {
+                case 1:
+                    this->unk_1EC = 2;
+                    break;
+                case 2:
+                    this->unk_1EC = 1;
+                    break;
+                case 3:
+                    this->unk_1EC = 7;
+                    break;
+                case 4:
+                    this->unk_1EC = 3;
+                    break;
+                case 5:
+                    this->unk_1EC = 6;
+                    break;
+                case 6:
+                    this->unk_1EC = 5;
+                    break;
+                case 7:
+                    this->unk_1EC = 4;
+                    break;
+                case 8:
+                    this->unk_1EC = 0;
+                    break;
+                case 10:
+                    this->unk_1EC = 8;
+                    break;
+                case 11:
+                    this->unk_1EC = 9;
+                    break;
+                case 13:
+                    this->unk_1EC = 0xA;
+                    break;
+                case 15:
+                    this->unk_1EC = 0xB;
+                    break;
+                case 16:
+                    this->unk_1EC = 0xC;
+                    break;
+                case 17:
+                    this->unk_1EC = 0xD;
+                    break;
+                case 18:
+                    this->unk_1EC = 0xE;
+                    break;
+                case 19:
+                    this->unk_1EC = 0x11;
+                    break;
+                case 20:
+                    this->unk_1EC = 0x12;
+                    break;
+                case 21:
+                    this->unk_1EC = 0x14;
+                    break;
+                case 22:
+                    this->unk_1EC = 0x15;
+                    break;
+                case 23:
+                    this->unk_1EC = 0x17;
+                    break;
+                case 24:
+                    this->unk_1EC = 0x18;
+                    break;
+                default:
+                    this->unk_1EC = 0;
+                    break;
+            }
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+        }
+
+        if ((this->unk_1EC == 5) && (globalCtx->sceneNum == 8) && (gSaveContext.sceneSetupIndex == 0xB) &&
+            (globalCtx->csCtx.frames == 0x190)) {
```

Comment:
```suggestion
            (globalCtx->csCtx.frames == 400)) {
```
frames can be in decimal